### PR TITLE
feat: add draft version of the scripts

### PR DESCRIPTION
**Pull request by ccoVeille**

### CODE REVIEW

**Comment by ccoVeille** [comment]
This is a very naive approach of the script I used to:
- rename my forks to contain the owner of the repository I forked
- move the forks from one owner to another or to a repository

That's how I moved everything to https://github.com/orgs/ccoveille-forks/repositories to let my https://github.com/ccoveille/?tab=repositories quite clean

**Comment by ccoVeille** [comment]
@ldemailly @mmorel-35 @yassinebenaid 

could you please review this ?

thanks

File: move-forks-to-organization.sh [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+if [ "$#" -ne 2 ]; then
+	name="$(basename "$0")"
+	printError "Usage: $name <user> <newowner>"
+fi
+
+user=$1
+newowner=$2
+
+which jq >/dev/null || printError "please install jq"
+which gh >/dev/null || printError "please install GitHub CLI"
```
Comment (yassinebenaid) [inline move-forks-to-organization.sh:19]: On some systems/shells, the `which` command will return 0 exit code regardless of the existance of the command.
Prefer using the `command` builtin instead..

```bash
command -v <CMD> >/dev/null || printError "..."
```

References:
- Bash docs: https://www.gnu.org/software/bash/manual/html_node/Bash-Builtins.html#index-command
- Stackoverflow:  https://stackoverflow.com/questions/592620/how-can-i-check-if-a-program-exists-from-a-bash-script

File: prefix-forks-with-owner.sh [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+}
+
+if [ "$#" -ne 1 ]; then
+	name="$(basename "$0")"
+	printError "Usage: $name <user>"
+fi
+user=$1
+
+which jq >/dev/null || printError "please install jq"
+which gh >/dev/null || printError "please install GitHub CLI"
```
Comment (yassinebenaid) [inline prefix-forks-with-owner.sh:17]: same as above.

File: prefix-forks-with-owner.sh [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+		printError "no name found for the repository"
+	fi
+
+	if [ -z "$owner" ]; then
+		printError "no owner found for $repo"
+	fi
+
+	if [ -z "$orepo" ]; then
+		printError "no original name found for $repo"
+	fi
```
Comment (yassinebenaid) [inline prefix-forks-with-owner.sh:32]: Can be shortened like you did above:

```bash
[ -z "$repo" ] && printError "no name found for the repository"

# ...etc
```

File: prefix-forks-with-owner.sh [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+	# skip rename if it's already contains the repo name
+	if [[ "$owner" =~ $repo ]]; then
+		continue
+	fi
+
+	# skip rename if it's already contains the owner name
+	if [[ "$repo" =~ $owner ]]; then
+		continue
+	fi
```
Comment (yassinebenaid) [inline prefix-forks-with-owner.sh:42]: Better you shorten this:
 
```bash
[[ "$owner" =~ $repo ]] && continue

# ...etc
```

**Review by yassinebenaid — CHANGES_REQUESTED** [review]
Interesing scripts, I'll need to try them out. I also don't like to have forks on my personal account.

File: move-forks-to-organization.sh (hunk 1 above)
Comment (ccoVeille) [reply move-forks-to-organization.sh:19]: Yez, indeed I codes quickly.

I'll iterate,that's why I opened a PR

File: prefix-forks-with-owner.sh (hunk 3 above)
Comment (ccoVeille) [reply prefix-forks-with-owner.sh:32]: Thanks, I nightly use the `test -z || printError ` syntax instead, mixing the short test syntax outside an if statement is strange to me

File: move-forks-to-organization.sh [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+gh auth status >/dev/null || printError "Please configure GitHub CLI with: gh auth login"
+
+gh repo list --fork --limit 1000 --json name,parent "$user" | jq -r '.[] | [.name] | @tsv' | while read -r repo; do
+	if [ -z "$repo" ]; then
+		printError "no name found for the repository"
+	fi
+
+	echo "$repo would be moved to $newowner with the same repo name"
```
Comment (ldemailly) [inline move-forks-to-organization.sh:28]: check it doesn't already exist in the destination?

File: prefix-forks-with-owner.sh [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+	printError "Usage: $name <user>"
+fi
+user=$1
+
+which jq >/dev/null || printError "please install jq"
+which gh >/dev/null || printError "please install GitHub CLI"
+
+gh auth status >/dev/null || printError "Please configure GitHub CLI with: gh auth login"
+
+gh repo list --fork --limit 1000 --json name,parent "$user" | jq -r '.[] | [.name, .parent.owner.login, .parent.name] | @tsv' | while read -r repo owner orepo; do
```
Comment (ldemailly) [inline prefix-forks-with-owner.sh:21]: nbd but the 2 scripts are essentially copy paste of the same start, maybe make one a subcommand or flag?

**Comment by mmorel-35** [comment]
Have you considered creating a go module to handle this ?
You'd be able to use credentials (gh auth) thought file with [viper](https://github.com/spf13/viper)
And use [go-github](https://github.com/google/go-github) to exchange with github API and follow the model

**Comment by ccoVeille** [comment]
@mmorel-35 I do, here I made a quick version with shell.

I opened

- #2

File: move-forks-to-organization.sh (hunk 5 above)
Comment (ccoVeille) [reply move-forks-to-organization.sh:28]: Maybe, but it's a bit out of the scope, the gh binary would simply report an error. So I think it's enough.

File: prefix-forks-with-owner.sh (hunk 6 above)
Comment (ccoVeille) [reply prefix-forks-with-owner.sh:21]: Yes, that's right.

I created one by copy pasting the first one 😅😬

I'm unsure what I could do now.

Maybe I will merge as this, after fixing some feedbacks.

**Comment by ccoVeille** [comment]
Thank you everyone.

I will iterate on a new version in shell at first.

I might go to a version in Go as a gh cli subcommand, as I suggested here #2

My main goal for this PR was to see if there was somehow a need for such tool.

I think the few feedbacks I'm getting from you, seem to confirm.

File: move-forks-to-organization.sh (hunk 1 above)
Comment (ccoVeille) [reply move-forks-to-organization.sh:19]: I will merge now, but I will address your feedback with:

- https://github.com/ccoVeille/github-forks-sanitizer/issues/4

File: prefix-forks-with-owner.sh (hunk 3 above)
Comment (ccoVeille) [reply prefix-forks-with-owner.sh:32]: I will merge now, but I don't think I will address this stylistic remark.

Feel free to open a PR, if you think I should

File: prefix-forks-with-owner.sh (hunk 6 above)
Comment (ccoVeille) [reply prefix-forks-with-owner.sh:21]: I will address this with a complete rewrite

https://github.com/ccoVeille/github-forks-sanitizer/issues/2

File: prefix-forks-with-owner.sh (hunk 4 above)
Comment (ccoVeille) [reply prefix-forks-with-owner.sh:42]: Same as

https://github.com/ccoVeille/github-forks-sanitizer/pull/1#discussion_r1881667128

File: prefix-forks-with-owner.sh (hunk 2 above)
Comment (ccoVeille) [reply prefix-forks-with-owner.sh:17]: see https://github.com/ccoVeille/github-forks-sanitizer/pull/1#discussion_r1881664804

**Comment by ccoVeille** [comment]
Thank you everyone for your feedbacks.

I want to merge my code, because:
- this PR is now old, 
- well, it works, even if it could be better
- I will address your feedbacks in other issues
- I might rewrite everything to provide a `gh extension` directly, see #2 
- This repo looks empty without merging this MR.

So thank again, but sorry for not addressing them yet.

**Pull request merged**